### ISSUE TITLE
[PORT] clean up activityhandler and teamsactivityhandler

### ISSUE THREAD

**Issue by tomlm** [microsoft/botbuilder-js#1836]
> Port this change from botbuilder-dotnet/master branch:
https://github.com/microsoft/botbuilder-dotnet/pull/3482

fixes #3446 

(duplicate code and new compiler warning introduced with sso fixes)

# Changed projects
* Microsoft.Bot.Builder
* Microsoft.Bot.Builder.Tests

**Comment by Stevenic** [comment]
Needs porting to JS although I don't completely agree with how you're doing the code to protect against sending multiple invoke responses. This protection should be done in the adapter if anything.

**Comment by johnataylor** [comment]
This is a dup of #1680

**Comment by johnataylor** [comment]
This is now an issue follow the SSO check-in. Its slightly different to the C# but the principle it the same: we want any additional Invoke handling we do in ActivityHandler to be consistent with the work we had already done in TeamsActivityHandler. Basically move the code up to the base class.

**Comment by Kaiqb** [comment]
@johnataylor - is this waiting on a review from someone?
